pkg/storage/system_intake_funding_source.go

```diff
@@ -0,0 +1,73 @@
+package storage
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/uuid"
+
+	"github.com/cmsgov/easi-app/pkg/appcontext"
+	"github.com/cmsgov/easi-app/pkg/models"
+)
+
+// UpdateSystemIntakeFundingSources clears and updates the funding sources of a system intake
+func (s *Store) UpdateSystemIntakeFundingSources(ctx context.Context, systemIntakeID uuid.UUID, fundingSources []*models.SystemIntakeFundingSource) ([]*models.SystemIntakeFundingSource, error) {
+	now := s.clock.Now()
+
+	tx := s.db.MustBegin()
+	defer tx.Rollback()
+	deleteFundingSourcesSQL := `
+		DELETE FROM system_intake_funding_sources
+		WHERE system_intake_id = $1;
+	`
+	_, err := tx.Exec(deleteFundingSourcesSQL, systemIntakeID.String())
+
+	if err != nil {
+		appcontext.ZLogger(ctx).Error(fmt.Sprintf("Failed to create funding sources transaction, error %s", err))
+		return nil, err
+	}
+
+	for _, fundingSource := range fundingSources {
+		if fundingSource != nil {
+			if fundingSource.ID == uuid.Nil {
+				fundingSource.ID = uuid.New()
+			}
+			if fundingSource.CreatedAt == nil {
+				fundingSource.CreatedAt = &now
+			}
+			const createFundingSourceSQL = `
```

Comment:
Incidentally, sqlx is supposed to support batch insert of structs, but it looks like there's a bug with the version of it (1.2.0) we have - see https://github.com/jmoiron/sqlx/issues/519. It's not worth worrying about for this PR (batch inserting would save a few database round-trips, but that's probably pretty minor), but it'd be nice to bump the sqlx version at some point so we can use batch inserts.